src/com/facebook/buck/android/SmartDexingStep.java

```diff
@@ -463,9 +471,20 @@ static void createDxStepForDxPseudoRule(
       OptionalInt xzCompressionLevel,
       Optional<String> dxMaxHeapSize,
       String dexTool,
-      @Nullable Collection<Path> classpathFiles) {
+      @Nullable Collection<Path> classpathFiles,
+      boolean useDexBuckedId) {
 
+    Optional<String> buckedId = Optional.empty();
     String output = outputPath.toString();
+    if (useDexBuckedId) {
+      // We know what the output file name is ("classes.dex" or "classesN.dex") as these
+      // are generated in SplitZipStep and passed around as part of a multi-map - it is
+      // simply easier and cleaner to extract the dex file number to be used as unique
+      // identifier rather than creating another map and pass it around
+      String[] tokens = outputPath.getFileName().toString().split("\\.")[0].split("classes");
```

Comment:
You can use `Files.getNameWithoutExtension()` instead.

Reply:
Done